### DISCUSSION TITLE
Nematode genome non-circularized

### DISCUSSION THREAD

**Discussion by desmodus1984** [Kinggerm/GetOrganelle#323] [Q&A]
Dear Dr. Jin,

Hope you are doing well. I am contacting you because I am working with nematodes, and I just recently found that their mitogenomes are very rearranged. I have sequenced two new families, and I have also used raw sequences with higher depth from a colleague working with C. elegans, and I was able to get the mitogenome, but while visualizing it with bandage, not of them are circular/ized. 
![C ele 117](https://github.com/Kinggerm/GetOrganelle/assets/40648021/ba60e940-acee-4647-a65d-d4515ab1dfb7)

[117.get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/14797265/117.get_org.log.txt)
[763.127.get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/14797266/763.127.get_org.log.txt)

Also, once I read, that merging reads, can benefit genome assembly stats, higher contiguity/less contigs. Thus, I tried merging PE-reads, with one of my species and got a very weird graph. It was very chaotic when I expected that the longer reads might help getting the circular genome.
Raw-reads
[3391.get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/14797282/3391.get_org.log.txt)
![3391 115](https://github.com/Kinggerm/GetOrganelle/assets/40648021/27b88650-5ebe-4bdc-b0de-b2ccd4a9e73d)

Merged reads
[3391.mrg.get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/14797281/3391.mrg.get_org.log.txt)
![3391 mrg graph](https://github.com/Kinggerm/GetOrganelle/assets/40648021/e8c98fcc-9fcb-40d6-8fdb-3cc1761a05e3)

As you can see, with the merged reads, I even get a partially assembled contig, instead of an unique one. Any hint of why this behavior is happening?

I would appreciate if you could tell if you have tried a nematode before, and have succeeded getting a circular contig, and in my case, one species, the coverage isn't that high ~20X, but I was still able to get a single contig like the 117, the other species is higher (messier graph), which parameter I could try to optimize to get a circular mitogenome.

Thank you very much.

Juan Pablo

**Comment by JianjunJin** [comment]
Hi Juan,

Thank you for reaching out!

To ensure we're on the same page, could you please specify the file names you're visualizing when referring to the graphs? This will help clarify the context and allow for more accurate support.

If a mitogenome with sufficient depth (aside from the 763 sample) does not circularize, it may be due to complex repeat structures within the genome. In such cases, GetOrganelle tends to export only the regions that are free of unresolved repeats, causing the connecting repeats to be chopped off. You should probably see them when you visualize the `*.fastg` file.

**Comment by desmodus1984** [comment]
Hi Dr. JianJun,

Thanks for your quick response.

It is my pleasure to help you clarify my case. The first graph is an attempt to assemble the mitogenome of Caenorhabditis elegans, it is 117 (C.ele.117. png).
It is a high-depth sequencing, which I used standard parameters, and as seen, Bandage visualization didn't show any repeat, like it has shown before when trying a mitogenome of a bat, or another nematode mitogenome (species 763 using standard parameters - max kmer 115).
[cid:41a65828-cb94-4c1d-8d2d-f648f79be8f3]
In this species, 763, when using the highest kmer value, 127, since I have 150 bp reads, that "repeat" disappeared:
[cid:89a62f26-f375-4a15-9363-53224069284f]
and the final fasta went from 12457 to 12469.
And as similar, with the C. elegans attempt, I don't get that "repeat" in red, and the genome is linear, and not circular.
[cid:5a559adf-e5f3-46a9-811e-0738041d4bb4]

Then, I tried the same with species 3391, using a longer kmer but the results didn't improve.

Next, I read that merging paired-end reads benefits genome assembly
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0185056
[https://journals.plos.org/plosone/article/figure/image?id=10.1371/journal.pone.0185056.g006&size=inline]<https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0185056>
BBMerge – Accurate paired shotgun read merging via overlap<https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0185056>
journals.plos.org

So I thought that, with merged reads, perhaps it could benefit, make circularization more likely, but it didn't help much for species 763, but for 3391, the graph turned into a mess with several fragments, instead of the single one I got when using the trimmed unmerged reads:
With unmerged reads:[cid:129751da-2927-43d3-b11e-32aa93a8b76f]
(3391.115.png)
But assembly with the merged reads (3391.mrg.graph.png) it turned into a convoluted knot, with even some "repeat regions", which didn't appear when assembling using unmerged reads.
[cid:7179d232-d26c-4c57-af51-0bd9d1c8e818]

I thought that ok, for my species I might not be able to assemble a circular mitogenome because of depth, but that didn't occur with C elegans which had a depth of 232X.

I know that nematode mitogenome tend to rearrange a lot, but I don't understand why I get a circular for my longer bat mitogenome:
[cid:f1d2a80a-0617-4e0c-af16-30b072891a60]
but I get a linear for nematode.

I am eager to hear your comments and thoughts.

Sincerely & Thank you very much.



Juan Pablo Aguilar Cabezas

________________________________
From: JianJun Jin ***@***.***>
Sent: Tuesday, April 2, 2024 3:34 PM
To: Kinggerm/GetOrganelle ***@***.***>
Cc: Aguilar Cabezas, Juan Pablo ***@***.***>; Author ***@***.***>
Subject: [External] Re: [Kinggerm/GetOrganelle] Nematode genome non-circularized (Discussion #323)

Use caution with links and attachments.

Hi Juan,

Thank you for reaching out!

To ensure we're on the same page, could you please specify the file names you're visualizing when referring to the graphs? This will help clarify the context and allow for more accurate support.

If a mitogenome with sufficient depth (aside from the 763 sample) does not circularize, it may be due to complex repeat structures within the genome. In such cases, GetOrganelle tends to export only the regions that are free of unresolved repeats, causing the connecting repeats to be chopped off. You should probably see them when you visualize the *.fastg file.

**Comment by JianjunJin** [comment]
Let me be clear, please always specify the name of the graph file (not the sample) you loaded in Bandage along with each png image.

BTW, I cannot see the image from your latest comment

**Comment by desmodus1984** [comment]
Let me try to address your comment.
I loaded in bandage the fastg (and csv) from the longest kmer, located in the K115 or K127 folder in the extended_spades directory.


Juan Pablo Aguilar Cabezas

Ecology and Evolutionary Biology Ph.D. Candidate

Department of Biological Sciences

Ohio University, Athens OH

________________________________
From: JianJun Jin ***@***.***>
Sent: Tuesday, April 2, 2024 9:12 PM
To: Kinggerm/GetOrganelle ***@***.***>
Cc: Aguilar Cabezas, Juan Pablo ***@***.***>; Author ***@***.***>
Subject: [External] Re: [Kinggerm/GetOrganelle] Nematode genome non-circularized (Discussion #323)

Use caution with links and attachments.

Let me be clear, please always specify the graph file you loaded in Bandage along with each png image

**Reply by JianjunJin** [reply]
You were not loading the fastg file because the graph of [3391](https://github.com/Kinggerm/GetOrganelle/files/14797282/3391.get_org.log.txt) and others has multiple contigs according to the log file, while the image only contain one contig

**Reply by desmodus1984** [reply]
Hi,

I did load the graphs of 117 (C elegans) and 3391 (non-merged reads), and at least by the graphs, there was only one contig. The 3391 with merged reads, I loaded again, and It looked chaotic, . I am attaching the files of 3391-merged, in case you want to verify.
The logs of 3391 (non-merged reads) and that of 117 mentioned only 1 graph and 1 scaffold.

117:
2024-03-27 17:09:28,130 - INFO: Writing output ...
2024-03-27 17:09:28,135 - INFO: Writing PATH1 of animal_mt scaffold(s) to 117/animal_mt.K115.scaffolds.graph1.1.path_sequence.fasta
2024-03-27 17:09:28,136 - INFO: Writing GRAPH to 117/animal_mt.K115.contigs.graph1.selected_graph.gfa
2024-03-27 17:09:28,136 - INFO: Result status of animal_mt: 1 scaffold(s)

Maybe you are reading a different log file, or, I miss the different part of the log file mentioning that.

According to the log file, average animal_mt coverage and the average animal_mt base-coverage of 117 were 232.8 and 969.9, respectively. Aren't these conditions enough to reach a circular mitochondrial nematode genome?
I am surprised that the nematode data wasn't enough to get a circular mitochondrial genome, but due to contamination in my bat sample, I did assemble a circular human one with just 17.1X depth.

Could you suggest any parameter to change to achieve a circular mt-genome - if it is possible?

Thank you very much for your attention and time.
[3391.mrg-reads.assembly_graph.fastg.extend-animal_mt.zip](https://github.com/Kinggerm/GetOrganelle/files/14860018/3391.mrg-reads.assembly_graph.fastg.extend-animal_mt.zip)
[3391.mrg-reads.get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/14860020/3391.mrg-reads.get_org.log.txt)

**Reply by JianjunJin** [reply]
I see. I'm sorry for the misreading.
I visualized the graph you provided (3391.mrg-reads.assembly_graph.fastg.extend-animal_mt.fastg), manually cleaned it, and didn't see the clue for further improvement. I gut feeling is it's something weird with the data.
[manual.graph.gfa.txt](https://github.com/Kinggerm/GetOrganelle/files/14860941/manual.graph.gfa.txt)
![manual graph](https://github.com/Kinggerm/GetOrganelle/assets/39948588/af400109-75f8-44da-be00-8d90fa359304)
